src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/MicrosoftExtensionsLogging/MicrosoftLoggingWrapper.cs

```diff
@@ -53,52 +55,64 @@ private void RecordLogMessage(MethodCall methodCall, MEL.ILogger logger, IAgent
+                // Get the first logger in the array (logger.Loggers[0])
+                object firstLogger = loggers.GetValue(0);
 
-            // Get the scope provider from that logger (logger.Loggers[0].Logger.ScopeProvider)
-            var getScopeProviderFunc = _getScopeProvider ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<MEL.IExternalScopeProvider>(internalLogger.GetType(), "ScopeProvider");
-            var scopeProvider = getScopeProviderFunc(internalLogger);
+                // Get the internal logger (logger.Loggers[0].Logger)
+                var getLoggerPropertyFunc = _getLoggerProperty ??= firstLogger.GetType().GetProperty("Logger").GetValue;
+                object internalLogger = getLoggerPropertyFunc(firstLogger);
 
-            // Get the context data
-            var harvestedKvps = new Dictionary<string, object>();
-            scopeProvider.ForEachScope((scopeObject, accumulatedKvps) =>
-            {
-                if (scopeObject is IEnumerable<KeyValuePair<string, object>> kvps)
+                // Get the scope provider from that logger (logger.Loggers[0].Logger.ScopeProvider)
+                var getScopeProviderFunc = _getScopeProvider ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<MEL.IExternalScopeProvider>(internalLogger.GetType(), "ScopeProvider");
+                var scopeProvider = getScopeProviderFunc(internalLogger);
+
+                // Get the context data
+                var harvestedKvps = new Dictionary<string, object>();
+                scopeProvider.ForEachScope((scopeObject, accumulatedKvps) =>
                 {
-                    foreach (var kvp in kvps)
+                    if (scopeObject is IEnumerable<KeyValuePair<string, object>> kvps)
+                    {
+                        foreach (var kvp in kvps)
+                        {
+                            accumulatedKvps.Add(kvp.Key, kvp.Value);
+                        }
+                    }
+                    else if (scopeObject is KeyValuePair<string, object> kvp)
                     {
                         accumulatedKvps.Add(kvp.Key, kvp.Value);
                     }
-                }
-                else if (scopeObject is KeyValuePair<string, object> kvp)
-                {
-                    accumulatedKvps.Add(kvp.Key, kvp.Value);
-                }
-                // Possibly handle case of IEnumerable<KeyValuePair<object, object>>, etc (not now though)
-            }, harvestedKvps);
+                    // Possibly handle case of IEnumerable<KeyValuePair<object, object>>, etc (not now though)
+                }, harvestedKvps);
 
-            return harvestedKvps;
+                return harvestedKvps;
+            }
+            catch (Exception e)
+            {
+                agent.Logger.Log(Level.Debug, $"Unexpected exception while attempting to get context data. Context data is not supported for this logging framework. Exception: {e}");
```

Comment:
I would log this at warning so customers don't need to turn logs up when they wonder why their context isn't working